pkg/plugins/resources/shell/environments.go

```diff
@@ -0,0 +1,70 @@
+package shell
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+)
+
+type Environments []Environment
+
+func (e *Environments) ToStringArray() []string {
+	result := []string{}
+
+	for _, environment := range *e {
+		result = append(result, environment.String())
+	}
+
+	return result
+
+}
+
+func (e *Environments) isDuplicate() bool {
+	foundName := map[string]string{}
+	foundDuplicatedName := []string{}
+
+	for _, env := range *e {
+		if _, ok := foundName[env.Name]; ok {
+			foundDuplicatedName = append(foundDuplicatedName, env.Name)
+			continue
+		}
+		foundName[env.Name] = env.Value
+	}
+
+	if len(foundDuplicatedName) > 0 {
+		logrus.Warningf("duplicated environment variable found: [%q]\n", strings.Join(foundDuplicatedName, ","))
+		return true
+	}
+
+	return false
+
```

Comment:
```suggestion
	foundName := map[string]struct{}{}
	foundDuplicatedName := []string{}

	for _, env := range *e {
		if _, ok := foundName[env.Name]; ok {
			foundDuplicatedName = append(foundDuplicatedName, env.Name)
			continue
		}
		
		foundName[env.Name] = struct{}{}
	}

	if len(foundDuplicatedName) > 0 {
		logrus.Warningf("duplicated environment variable found: [%q]\n", strings.Join(foundDuplicatedName, ","))
		return true
	}

	return false
```

Nit: you can use `map[string]struct{}` when you don't care about the value of a map. This also saves memory.